docs/run_test/understand_results.md

```diff
@@ -0,0 +1,73 @@
+# Understand test results
+
+- [Overview](#overview)
+- [Intermediate results](#intermediate-results)
+- [Final results](#final-results)
+
+## Overview
+
+It's essential to understand the results after running tests. LISA has 7 kinds
+of test results in total: 3 of which are intermediate results, and 4 of which
+are final results, as explained in sections below. Each test case can and will
+be moved from one result to another but can never have two or more results at
+the same time. 
+
+![test_results](../img/test_results.png)
+
+## Intermediate results
+
+An intermediate result shows information of an unfinished test. It will show up
+when a test changes its state. If a test run terminates because of error or
+exception prior to running a test case, only the intermediate result will be provided.
+
+- **QUEUED**
+
+  QUEUED tests are tests that are created, and planned to run (but have not
+  started yet). They suggest that there are some tests waiting to be performed.
+
+  QUEUED tests will move forward to ASSIGNED.
+
+- **ASSIGNED**
+
+  ASSIGNED tests are tests that are assigned to an environment, and will start
+  to run once the environment is deployed. They suggest some environmental
+  setting up is going on. 
+
+  ASSIGNED tests will end with FAILED if the environment fails to deploy. They
+  will also move backward to QUEUED again if the environment does not fit the
+  requirement of the test.
+
+- **RUNNING**
+
+  RUNNING tests are tests that are in test procedure. 
+
+  RUNNING tests will end with one of the following final results.
+
+## Final results
+
+A final result shows information of a terminated test. It provides more valuable
+information than the intermediate result. It only appears in the end of a
+successful test run.
+
+- **FAILED**
+
+  FAILED tests are tests that did not finish successfully and terminated because
+  of failures like `LISA exceptions` or `Assertion failure`. You can use them to
+  trace where the problem was and why the problem happened.
+
+- **PASSED**
+
+  PASSED tests are tests that passed, or at least partially passed, with a
+  special `PASSException` that warns there are minor errors in the run but they
+  do not affect the test result.
+
+- **SKIPPED**
+
+  SKIPPPED tests are tests that did not start and would no longer run. They
+  suggest failure to meet some requirements in the environment involved with the
```

Comment:
"the environment" may be multiple one. One of skipping reason is, a queued case will try to match every created environment. If no env is matched, it will be skipped.